com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/detect/MtaMetadataExtractor.java

```diff
@@ -0,0 +1,23 @@
+package com.sap.cloud.lm.sl.cf.core.cf.detect;
+
+import java.util.ArrayList;
+
+import com.sap.cloud.lm.sl.cf.core.cf.detect.entity.MetadataEntity;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMta;
+
+public interface MtaMetadataExtractor<T extends MetadataEntity> {
+
+    public void extract(T entity, DeployedMta metadata);
+    
+    default void initMetadata(T entity, DeployedMta metadata) {
```

Comment:
Following the above comment in MtaMetadataEntityAggregator.java, remove this method